runtime/src/main/cpp/ObjectTraversal.hpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+#ifndef RUNTIME_OBJECT_TRAVERSAL_H
+#define RUNTIME_OBJECT_TRAVERSAL_H
+
+#include <functional>
+#include <type_traits>
+
+#include "Memory.h"
+#include "Natives.h"
+#include "Types.h"
+
+namespace kotlin {
+
+// TODO: Consider an iterator/ranges based approaches for traversals.
+
+template <typename F>
+void traverseObjectFields(ObjHeader* object, F process) noexcept(std::is_nothrow_invocable_v<F, ObjHeader**>) {
```

Comment:
So, apparently our wasm toolchain is not 17 enough. I don't really want to implement the entire `std::is_nothrow_invocable`, so I'll use a simpler check here.